src/languages/dos.js

```diff
@@ -12,11 +12,30 @@ export default function(hljs) {
     /^\s*@?rem\b/, /$/,
     { relevance: 10 }
   );
+
+  // for matching comments starting with ::
+  const COMMENT_2 = hljs.COMMENT(
+    /^::.*$/, /$/,
```

Comment:
The begin should start and the end should end, so I think you want:

start: `/^::/`
end : `/$/`

The end shouldn't be included in the begin.